tensorflow_encrypted/protocol/pond.py

```diff
@@ -218,14 +221,18 @@ def helper(v: tf.Tensor) -> 'PondPublicTensor':
 
     def define_private_input(
         self,
-        provider: InputProvider,
+        player: Union[str, Player],
+        data_fn: Callable,
```

Comment:
could we call it `inputter_fn` instead? think that's closer to the standard mpc terminology of "input provider"

Reply:
alternatively `provider_fn` but maybe that's less clear

Reply:
stronger typing also important here (users might try to pass functions returning numpy arrays otherwise)